appng-maven-plugin/src/main/java/org/appng/maven/plugin/mojo/AbstractGeneratorMojo.java

```diff
@@ -0,0 +1,73 @@
+package org.appng.maven.plugin.mojo;
+
+import java.io.File;
+import java.util.Arrays;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.sonatype.plexus.build.incremental.BuildContext;
+
+abstract class AbstractGeneratorMojo extends AbstractMojo {
+
+    /**
+     * the path to the source file
+     */
+    @Parameter(property = "filePath", required = true)
+    protected File filePath;
+
+    /**
+     * the fully qualified name of the target class to generate
+     */
+    @Parameter(property = "targetClass", required = true)
+    protected String targetClass;
+
+    /**
+     * the output-folder for the generated class
+     */
+    @Parameter(property = "outFolder", defaultValue = "target/generated-sources/constants", required = false)
+    protected File outfolder;
+
+    /**
+     * skips the execution
+     */
+    @Parameter(property = "skip", defaultValue = "false", required = false)
+    protected boolean skip;
+
+    @Component
+    protected BuildContext buildContext;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        if (skip) {
+            getLog().info("skipping " + getMessage());
```

Comment:
This does not skip the execution. Is there a return statement missing?

Reply:
Hi Claus.

Thanks for your hints. Please take a look at https://github.com/appNG/appng/pull/16/commits/1962f268f0e7f7837e5be9698a798288b182d6bd and https://github.com/appNG/appng/pull/16/commits/38519f823c91864109bd16e50e532fda174fa321

Thanks,
Oliver